### DISCUSSION TITLE
[Marketplace management] Custom service type

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#250] [Q&A] [answered]
### Which section?

Marketplace

### Which is the version?

v10.8.0

### Description

Hi guys,
I have a question on the marketplace elements management.
As reported in the public documentation, a marketplace element belongs to one of the supported types:
- plugin
- template
- example
- proxy
- application

But types can be configured in two parts of the marketplace element:
- **type** property (root level);
- **resources.services** property &rarr; every element has a `resources.services` JSON, with a `type` property inside.

Can the user customize one of the types above through the Console CMS?
We tried creating a custom `plugin` of type _My type_ and the service creation worked fine:
<img width="304" alt="Screenshot 2023-05-10 at 15 33 20" src="https://github.com/mia-platform/community/assets/28152586/e02c7977-d052-46cb-9b17-9d75609ccc2e">

Following the content of the `resources.services` JSON:
```json
{
  "services": {
    "my-service": {
      "type": "plugin",
      "name": "my-service",
      "description": "My service",
      "repositoryUrl": "",
      "dockerImage": "my-registry.cloud/my-service",
      "defaultEnvironmentVariables": [
        {
          "name": "LOG_LEVEL",
          "value": "{{LOG_LEVEL}}"
        },
        {
          "name": "HTTP_PORT",
          "value": "3000"
        }
      ],
      "containerPorts": [
        {
          "name": "http",
          "from": 80,
          "to": 3000,
          "protocol": "TCP"
        }
      ],
      "defaultResources": {
        "cpuLimits": {
          "min": "10m",
          "max": "100m"
        },
        "memoryLimits": {
          "min": "100Mi",
          "max": "300Mi"
        }
      },
      "defaultProbes": {
        "liveness": {
          "path": ""
        },
        "readiness": {
          "path": ""
        }
      },
      "defaultConfigMaps": [
        {
          "name": "my-cm",
          "mountPath": "/app/configs",
          "files": [
            {
              "name": "key",
              "content": "value"
            }
          ]
        }
      ]
    }
  }
}
```

**Comment by davidebianchi** [comment, marked as answer]
Type at the first level are the types shown on the website of the marketplace, instead type inside the resource are the types managed under the hood which handles differently the types of elements.
At the moment it should not be used a custom types, since it is not supported. Why do you need some custom types?

**Reply by frengooo** [reply]
> At the moment it should not be used a custom types

Are you talking about the first-level type?

Anyway, it's not strictly a need at the moment, it's just to know because a client could need it.
In the future, I think it could be useful to grant access to users on specific types only.
